pkg/controller/job/job_controller.go

```diff
@@ -1695,9 +1699,22 @@ func findConditionByType(list []batch.JobCondition, cType batch.JobConditionType
 
 func recordJobPodFinished(job *batch.Job, oldCounters batch.JobStatus) {
 	completionMode := completionModeStr(job)
-	diff := job.Status.Succeeded - oldCounters.Succeeded
+	var diff int
+
+	// Updating succeeded metric must be handled differently
+	// for Indexed Jobs to handle the case where the job has
+	// been scaled down by reducing completions & parallelism
+	// in tandem, and now a previously completed index is
+	// now out of range (i.e. index >= spec.Completions).
+	if isIndexedJob(job) {
+		diff = succeededIndexesFromJob(job.Status.CompletedIndexes, int(*job.Spec.Completions)).total() - succeededIndexesFromJob(oldCounters.CompletedIndexes, int(*job.Spec.Completions)).total()
```

Comment:
We could save the cost of parsing if CompletedIndexes didn't change.

Reply:
Done